docs/design/AgentRun-CLI-Design.md

```diff
@@ -0,0 +1,658 @@
+
+To streamline this process, we propose a dedicated Command Line Interface (CLI) that provides:
+- Lightweight orchestration of agent development workflow, including initialization, packaging, building, testing, and publishing;
+- Extensible architecture supporting local environments, Kubernetes clusters, and multi-cloud build and deployment targets;
+- Python SDK bindings for programmatic access to CLI functionality, nabling integration into CI workflows and continuous development pipelines.
+
+This CLI will serve as a foundational tool enabling:
+- Rapid prototyping and local iteration of agents
+- Seamless transitions from development to staging and production environments powered by AgentCube
+- Native integration with Kubernetes clusters and cloud services for scalable deployment
+- Version control and reproducibility of agent configurations and runtime environments
+
+By abstracting away operational overhead and providing a consistent interface, the CLI lowers the barrier to entry for agent development on AgentCube. It empowers developers to focus on innovation rather than infrastructure, fosters best practices, and accelerates collaboration across the open-source community.
+## Use Case 1: Kickstart Agent Development and Publish to AgentCube
+A developer wants to create a new agent from scratch and publish it to AgentCube for public access or team collaboration. The CLI supports agents built with any framework, offering a standardized workflow for packaging, building, and publishing. 
+
+After completing development, the CLI enables the following steps:
+1. `agentrun pack -f ./` Packages the agent source code and runtime metadata into a structured workspace directory, preparing it for image creation
+2. `agentrun build -f ./` Builds a container image from the workspace, compatible with AgentCube’s Kubernetes-based runtime environment
+3. `agentrun publish -f /` Publishes the built agent image to AgentCube, making it available for invocation, sharing, and collaboration
+
+Use Case 2: Check Published Agent Status
+After publishing an agent to AgentCube, a developer may want to verify that the agent is fully registered and ready for use. The CLI provides a simple status check command:
+
+```agentrun status -f ./```
+
+This command queries AgentCube for the current state of the agent associated with the workspace. It returns key information such as the agent ID, endpoint URL, latest version, and log reference. This helps ensure the agent is correctly deployed and ready for invocation.
+
+## Use Case 3: Invoke published Agent
+After publishing an agent to AgentCube, a developer may want to invoke it for testing purposes or integrate it into other system components. The CLI provides a simple and consistent interface to trigger agent execution using the local workspace directory and a structured payload:
+
+```agentrun invoke -f ./ -payload {"prompt": "what is the weather today in Shanghai?"}```
```

Comment:
same as above, supersed -f with some id.

Reply:
As mentioned above, I don’t think the ID alone is sufficient. We also need the URL, which varies across providers and authentication methods. Since there are too many options to manage directly, I use metadata within the workspace to handle these configurations.